### ISSUE TITLE
CircleCI : use only Status permission for the API calls

### ISSUE THREAD

**Issue by Horgix** [badges/shields#1064]
**tl;dr : I think the CircleCI badge with token should be able to work with an API token having the "Status" scope only**

When generating an API token for CircleCI (used for private projects badges), it is possible to assign a scope to it. They currently support 3 scopes : "Status", "Build artifacts" and "all".

"Status" scope should be enough to get informations for a badge, and it is the scope used for the "official" CircleCI tokens. Yet, shields.io requires a "all" scope, and will display a "permission denied" badge otherwise. I think that it could and should be limited to API calls allowed by the "Status" permission since it's made for it; and I am a bit concerned about using "all" scope for a badge since it gives full access to the API on a private project...

**Comment by Horgix** [comment]
Any opinion on that?

If you tell me that a PR fixing this would get merged I might implement it 🙂

**Comment by paulmelnikow** [comment]
Hi! This sounds like an excellent idea. If you can work up a pull request and add some tests I would be really happy to review and merge it.

**Comment by Horgix** [comment]
It looks like there is no endpoint on CircleCI that is able to make use of this restricted scope for now 🙁 
I opened a Feature Request to CircleCI here : ["CircleCI / Get branch status with status-only scoped API token "](https://discuss.circleci.com/t/get-branch-status-with-status-only-scoped-api-token/18535)

I'll follow up here when I get some news!

**Comment by RedSparr0w** [comment]
There was a little bit of talk about this awhile ago [here](https://github.com/badges/shields/pull/798#issuecomment-255618281), not sure if anything has changed on CircleCI's end since then though.

**Comment by Horgix** [comment]
Thanks for the input @RedSparr0w , much appreciated 👍 
I can confirm that it works with the "Build Artifacts" scope; however I still think that it would be best to be able to use only a "Status" scoped token, do you agree?

I think I'll let this open until I get some news from CircleCI about the request of a new endpoint being able to work with a status-only scoped token

**Comment by paulmelnikow** [comment]
Huh. How come doesn't the status token doesn't work for our badge?

**Comment by RedSparr0w** [comment]
The status token doesn't seem to do anything from what i can tell, always get a `permission denied` message from the API with the `status` token.
```json
{
  "message" : "Permission denied"
}
```
Haven't found any endpoint which accepts it.
I hope its just an oversight on CircleCI's end but not sure.

**Comment by chris48s** [comment]
I've not tried, but if there's a SVG badge we can probably scrape it using the `BaseSvgScrapingService` class. If that gives us a status value which aggregates all the builds in a workflow, that would indeed make all the faffing about I've proposed in #1995 pointless (which would be a good thing).

**Comment by paulmelnikow** [comment]
The Circle route will need to be updated for #3329.

While Shields has long had a policy of not accepting secrets on the public badge service, this service has been an exception. Perhaps we could resolve the policy conflict by solving this issue. If the badge is using a status-only token, we're in fine shape.

It should fix #1872 and #1495 at the same time.
